packages/react-data-grid/src/HeaderGroupRow.js

```diff
@@ -0,0 +1,124 @@
+const React             = require('react');
+const createReactClass = require('create-react-class');
+const shallowEqual    = require('fbjs/lib/shallowEqual');
+const HeaderGroupCell        = require('./HeaderGroupCell');
+const getScrollbarSize  = require('./getScrollbarSize');
+const ExcelColumn  = require('./PropTypeShapes/ExcelColumn');
+const ColumnUtilsMixin  = require('./ColumnUtils');
+const SortableHeaderCell    = require('./cells/headerCells/SortableHeaderCell');
+const FilterableHeaderCell  = require('./cells/headerCells/FilterableHeaderCell');
+const HeaderCellType = require('./HeaderCellType');
+const createObjectWithProperties = require('./createObjectWithProperties');
+require('../../../themes/react-data-grid-header.css');
+
+import PropTypes from 'prop-types';
+
+const HeaderRowStyle  = {
+  overflow: PropTypes.string,
+  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  height: PropTypes.number,
+  position: PropTypes.string
+};
+
+// The list of the propTypes that we want to include in the HeaderRow div
+const knownDivPropertyKeys = ['width', 'height', 'style', 'onScroll'];
+
+const HeaderGroupRow = createReactClass({
+  displayName: 'HeaderGroupRow',
+
+  propTypes: {
+    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    height: PropTypes.number.isRequired,
+    groups: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
+    style: PropTypes.shape(HeaderRowStyle),
+    onScroll: PropTypes.func,
+    rowType: PropTypes.string,
+    draggableHeaderCell: PropTypes.func,
+    onHeaderDrop: PropTypes.func
+  },
+
+  mixins: [ColumnUtilsMixin],
```

Comment:
Lets remove this mixin  - it will prevent us from upgrading to React 16. Instead import any functions that you need from columnUtilsMixin and use them in your component. If the function is dependent on this.props, or this.state then refactor and pass them in as props